packages/hash/frontend/src/blocks/page/BlockContextMenu/LoadEntityMenuContent.tsx

```diff
@@ -0,0 +1,169 @@
+import { faAsterisk, faSearch } from "@fortawesome/free-solid-svg-icons";
+import { isBlockEntity } from "@hashintel/hash-shared/entityStore";
+import {
+  addEntityStoreAction,
+  entityStorePluginStateFromTransaction,
+} from "@hashintel/hash-shared/entityStorePlugin";
+import {
+  Box,
+  InputAdornment,
+  ListItemIcon,
+  ListItemText,
+  MenuItem,
+  MenuList,
+} from "@mui/material";
+import { PopupState } from "material-ui-popup-state/core";
+import { useCallback, useEffect, useRef, VFC } from "react";
+import { BlockEntity } from "@hashintel/hash-shared/entity";
+import { useBlockView } from "../BlockViewContext";
+import { FontAwesomeIcon } from "../../../shared/icons";
+import { useRouteAccountInfo } from "../../../shared/routing";
+import { LoadingSpinner, TextField } from "../../../shared/ui";
+import { useAccountEntities } from "../../../components/hooks/useAccountEntities";
+
+type LoadEntityMenuContentProps = {
+  entityId: string | null;
+  popupState?: PopupState;
+};
+
+export const LoadEntityMenuContent: VFC<LoadEntityMenuContentProps> = ({
+  entityId,
+  popupState,
+}) => {
+  const { accountId } = useRouteAccountInfo();
+  const { data: entities, fetchEntities, loading } = useAccountEntities();
+  const blockView = useBlockView();
+  const searchInputRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    if (popupState?.isOpen) {
+      searchInputRef.current?.focus();
+    }
+  }, [popupState]);
+
+  useEffect(() => {
+    const entityStore = entityStorePluginStateFromTransaction(
+      blockView.view.state.tr,
+      blockView.view.state,
+    ).store;
```

Comment:
This is depending on state external to React without explicitly subscribing to it. That can cause bugs. I don't think it's particularly relevant here, but we should add a todo to make this a subscription (perhaps with a `useEntity` hook).

Reply:
Agree. Added a comment for it